### ISSUE TITLE
Rx from Ti -> Arduino doesn't seem to work

### ISSUE THREAD

**Issue by ikt32** [fingalo/serial2ti83#1]
Thanks for responding so quickly! I've pasted the original message here below.

> I've been trying to communicate with my Ti-83+ using this sketch. jw0k's variant works, but failed to transfer larger files so I tried my luck with these forks. I just can't get them to work, however. I've tried it on the Arduino Uno and Nano, checked the cables, but things just don't seem to work. It looks like the calculator doesn't send anything back, as the TX light stays off.
>
> With the fork from DSchndr, the TX light does light up when manually sending from the calculator, though the calculator doesn't seem to respond to the bytes sent from the PC.
>
> With this fork, the last additional checks in getByte - always times out.

Your question:

> have you changed the connection to use A0 and A1 when using this and fingalo code for uno board?

Yes, I've plugged my cables into A0 and A1.

Can a long 2.5mm -> 3.5mm cable cause for the signal to deteriorate?

Does battery level impact stability? I haven't seen signs of low battery, but they've been in there for quite a while.

**Comment by fingalo** [comment]
Hi,
I checked the code here again and it is working on my UNO board, using the Texas connect software.
Your setup is working with jw0k's code which indicate your setup is OK.
Battery could be an issue, think I've seen that while testing earlier, I've used a LOT of batteries during my testing. Long cable could be an issue also as the code now is much faster with direct pin i/o.
Somebody suggested to use resistors (about 1 kohm) in series with the cable as they found out the Arduino board could not drive the calculator lines correctly. 
I have not experienced that in my testing and I've used several boards like Leonardo, UNO , Nano , Pro Micro.

I have also done  the changes recommended in the documentation for the library
#define SERIAL_TX_BUFFER_SIZE 64
#define SERIAL_RX_BUFFER_SIZE 64
to:
#define SERIAL_TX_BUFFER_SIZE 256
#define SERIAL_RX_BUFFER_SIZE 256

regards
fingalo

**Comment by ikt32** [comment]
Hmm, got some new batteries but the results are the same.

1kOhm resistors doesn't seem to improve reliability.

Buffer size also didn't seem to improve anything.

I threw in some more debugging stuff (different blinking rates) to figure out where it goes wrong - Transmit (from the Uno) seems to work fine, without any buffering or timeout issues.

Cursor on the calculator does blink more slowly while the Arduino is sending stuff to it, indicating it sees *something*, but there's no reply back from the calculator it seems. Or at least, it fails this check:

```cpp
    // only enter if a pin is low, otherwise you always get the timeout 
    if (TI_RING_IS_HIGH && TI_TIP_IS_HIGH) return false;
```

All I can change about my setup now is changing the cable, but this is the only 2.5mm cable I have. What length of 2.5mm cable do you use/have?

**Comment by fingalo** [comment]
Hi,
my cable is short, about 20 cm.
It seem that A0, or A1 port does not detect the low state when communicating.
I think the low level should be less than about 3 V to be detected as low level.

Could you check if the port is working? Like connect A0 and A1 to ground and read the value and write it to the console?

I can't remember if I have made any other changes to get it to work, but I don't think so.

**Comment by fingalo** [comment]
Hello, I've made a version of my code to use pin 8 as tip and pin 9 as ring (for UNO board).

Also when testing I noted problems when connecting cable. Some times the calculator becomes very slow, pressing digits response is veryyyy slow, and does not respond to transfer. I took out batteries, put them back, connected cable and then turned on the calculator. Then it worked again.
I tested both with TI connect and TILP softwares.

I have two TI83+, both working with this code. 

[serial2ti83_gen_fast.zip](https://github.com/fingalo/serial2ti83/files/5474675/serial2ti83_gen_fast.zip)
